### PR TITLE
Study nurse form

### PR DESCRIPTION
**Pull request by mustiwebp**
The study nurse form for additional entries has been implemented.

### CODE REVIEW

**Review by S1rDan — CHANGES_REQUESTED** [review]
Sorry, but there are a lot of issues with this one.

File: lib/screens/study_nurse_screen/configuration_screen/pages/configuration_additional_entries.dart [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+              },
+              hint: AppLocalizations.of(context)!.instituteKey,
+              validator: (String? value) {
+                if (value == null || value.isEmpty) {
+                  return '''     ${AppLocalizations.of(context)!.enterInstituteKey}''';
+                }
+                return null;
+              },
+            ),
+          ),
```
Comment (S1rDan) [inline lib/screens/study_nurse_screen/configuration_screen/pages/configuration_additional_entries.dart:103]:
```suggestion
          PicosSelect(
            selection: const <String>[
              '100',
              '101',
              '102',
              '103',
              '104',
              '105',
              '201',
              '300',
              '400',
              '501',
              '502'
            ],
            callBackFunction: (String? value) {
              widget.callbackAdditionalEntries('entryInstituteKey', value!);
            },
            hint: AppLocalizations.of(context)!.instituteKey,
            validator: (String? value) {
              if (value == null || value.isEmpty) {
                return '''     ${AppLocalizations.of(context)!.enterInstituteKey}''';
              }
              return null;
            },
          ),
```
I've tested, that this SizedBox is unnecessary now.

File: lib/screens/study_nurse_screen/configuration_screen/pages/configuration_form.dart [hunk 2]
```diff
@@ -154,12 +155,12 @@ class _ConfigurationFormState extends State<ConfigurationForm> {
...
-                IsRequired(AppLocalizations.of(context)!.entryEmail),
-                IsEmail(AppLocalizations.of(context)!.entryValidEmail),
-              ],
-            ),
+            validator: (String? value) {
+              if (value == null || value.isEmpty) {
+                return AppLocalizations.of(context)!.entryEmail;
+              }
+              return null;
+            },
```
Comment (S1rDan) [inline lib/screens/study_nurse_screen/configuration_screen/pages/configuration_form.dart:163]: Here, you aren't checking for a valid e-mail anymore. Is that intended?

File: lib/widgets/picos_select.dart [hunk 3]
```diff
@@ -60,49 +64,64 @@ class _PicosSelectState extends State<PicosSelect> {
...
+                final String value = newValue ?? '';
+
+                widget.callBackFunction(value);
+                _dropdownValue = value;
+              });
+            },
+            items: _createItemList(),
+            validator: widget.validator,
           ),
         ),
```
Comment (S1rDan) [inline lib/widgets/picos_select.dart:126]:
```suggestion
        child: ButtonTheme(
          alignedDropdown: true,
          child: SizedBox(
            height: widget.validator == null ? null : 75,
            child: DropdownButtonFormField<String>(
              decoration: InputDecoration(
                contentPadding: const EdgeInsets.only(
                  right: 10,
                ),
                enabledBorder: outlineInputBorder,
                focusedBorder: outlineInputBorder,
                errorBorder: errorInputBorder,
                focusedErrorBorder: errorInputBorder,
              ),
              borderRadius: borderRadius,
              value: _dropdownValue,
              icon: const Icon(Icons.keyboard_arrow_down),
              hint: Text(widget.hint ?? ''),
              onChanged: (String? newValue) {
                setState(() {
                  final String value = newValue ?? '';

                  widget.callBackFunction(value);
                  _dropdownValue = value;
                });
              },
              items: _createItemList(),
              validator: widget.validator,
            ),
          ),
        ),
```
Small improvement. If validator is set the it makes automatically a small space for the error message, so the gaps between all imput elements look similar.

File: lib/models/patient_data.dart [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+  /// denotes the discharge reason.
+  final String? dischargeReason;
+
+  /// denotes the start time of ICU stay.
+  final DateTime? azpICU;
+
+  /// denotes the mortality rate within hospital.
+  final double? khMortality;
+
+  /// denotes the lgbt70 value.
```
Comment (S1rDan) [inline lib/models/patient_data.dart:88]:
```suggestion
  /// denotes the LBgt70 value.
```

File: lib/models/patient_data.dart [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+  final bool? lgbt70;
+
+  /// denotes the list of icd10 codes.
```
Comment (S1rDan) [inline lib/models/patient_data.dart:91]:
```suggestion
  /// denotes the list of ICD 10 codes.
```

File: lib/models/patient_data.dart [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+  /// denotes the ICU mortality.
+  final double? icuMortality;
+
+  /// denotes the length of the ICU stay.
+  final int? icuLengthStay;
+
+  /// denotes the length of the stay in hospital.
+  final int? khLengthStay;
+
+  /// denotes the wdaKH value.
```
Comment (S1rDan) [inline lib/models/patient_data.dart:118]:
```suggestion
  /// denotes the WdaKH value.
```

File: lib/models/patient_data.dart [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+  final double? wdaKH;
+
+  /// denotes the weznDisease value,
```
Comment (S1rDan) [inline lib/models/patient_data.dart:121]:
```suggestion
  /// denotes the WEZnDisease value.
```

**Review by S1rDan — CHANGES_REQUESTED** [review]
Some more litte things.
I also suggested changes for some comments in the model because acronyms should be shown case sensitive imo.

File: lib/screens/study_nurse_screen/configuration_screen/pages/configuration_form.dart (hunk 2 above)
Comment (mustiwebp) [reply lib/screens/study_nurse_screen/configuration_screen/pages/configuration_form.dart]: No, it wasn't intended. It has been added now.

File: lib/screens/study_nurse_screen/configuration_screen/pages/configuration_form.dart [hunk 8]
```diff
@@ -154,12 +155,17 @@ class _ConfigurationFormState extends State<ConfigurationForm> {
...
+            validator: (String? value) {
+              if (value == null || value.isEmpty) {
+                return AppLocalizations.of(context)!.entryEmail;
+              } else if (!RegExp(
+                r"^[a-zA-Z0-9.a-zA-Z0-9.!#$%&'*+-/=?^_`{|}~]+@[a-zA-Z0-9]+\.[a-zA-Z]+",
+              ).hasMatch(value)) {
+                return AppLocalizations.of(context)!.entryValidEmail;
+              } else {
+                return null;
+              }
```
Comment (S1rDan) [inline lib/screens/study_nurse_screen/configuration_screen/pages/configuration_form.dart:167]:
```suggestion
              if (value == null || value.isEmpty) {
                return AppLocalizations.of(context)!.entryEmail;
              } else if (!RegExp('.+@.+',).hasMatch(value)) {
                return AppLocalizations.of(context)!.entryValidEmail;
              } else {
                return null;
              }
```
Regex is too difficult and doesn't even include some valid cases.
A simple regex as that should suffice.

**Review by S1rDan — CHANGES_REQUESTED** [review]

**Review by S1rDan — APPROVED** [review]

**Pull request merged**